readthedocs/core/resolver.py

```diff
@@ -112,28 +111,13 @@ def resolve_path(
 
         filename = self._fix_filename(project, filename)
 
-        current_project = project
-        project_slug = project.slug
-        subproject_slug = None
-        # We currently support more than 2 levels of nesting subprojects and
-        # translations, only loop twice to avoid sticking in the loop
-        for _ in range(0, 2):
-            main_language_project = current_project.main_language_project
-            relation = current_project.get_parent_relationship()
-
-            if main_language_project:
-                current_project = main_language_project
-                project_slug = main_language_project.slug
-                language = project.language
-                subproject_slug = None
-            elif relation:
-                current_project = relation.parent
-                project_slug = relation.parent.slug
-                subproject_slug = relation.alias
-                cname = relation.parent.domains.filter(canonical=True).first()
-            else:
-                break
-
+        main_project, subproject_slug = self._get_canonical_project_data(project)
+        project_slug = main_project.slug
+        cname = (
+            cname
+            or self._use_subdomain()
+            or main_project.get_canonical_custom_domain()
+        )
```

Comment:
I thought we were making use of the real `cname`, but we only use it as a boolean value to see if we need to serve from `/docs` instead of `/`.